src/Platform.Handlers/src/Xamarin.Platform.Handlers/Core/ISwitch.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Xamarin.Forms;
+
+namespace Xamarin.Platform
+{
+	public interface ISwitch : IView
+	{
+		bool IsToggled { get; set; }
+		Color OnColor { get; }
+		Color ThumbColor { get; }
```

Comment:
Does there need to be a default/off color or is this just background color?

Reply:
Yea there's a conversation here about the OffColor

https://github.com/xamarin/Xamarin.Forms/pull/10355

I'm still a fan of just having a property called "TrackColor" and then we just get rid of OnColor

There's some conversation in the issue above around that.

We can still leave oncolor and offColor inside Forms but on the Maui side use TrackColor

Reply:
I see. Because of the wait the native things work - having colors fade and transition under the hood - it might make sense to keep those in the handler. There is talk of switching to brushes, but is it supported on any of the platforms? And also, is the VSM at the handlers level or just Controls? Not sure if Comet has a concept of VSM?

Reply:
As far as I remember Comet has something similar to VSM where you can define the state/color relationship

>  is the VSM at the handlers level or just Controls

Currently just at the Controls level. We had talked about this a little bit a couple months ago and the approach I've taken here is kind of the approach we were curious about. Could we just set the colors based on state and the handlers would then just "learn". So, the way I built this if the user is toggling on VSM then the ColorStateList will eventually match.  This does take one iteration to work but maybe we could eventually project the VSM over the handler on init? If we have a Controls.SwitchHandler that knows about VSM then that's a possible scenario.

I realize the comments on the Switch were saying to keep both because of native transitions but this same idea kind of extrapolates out across every single control right? Pretty much all the coloring APIs on iOS/UWP/Android do it via some sort of state based coloring. This isn't unique to switch

Reply:
@mattleibow even if we don't want to be smart about matching ColorStateLists to current states it's just nicer in general to use ColorStateLists for setting the color for all states. If I want red I just set a CSL once for all the state attributes and then it just works.

When using the color filter you have to call "SetColorFilter" every time the color changes
https://github.com/xamarin/Xamarin.Forms/pull/12196/files#diff-5f11a6818b5ece8c3e67df6b8e71a734267ac65f49c428525efb61ec8122a08bR74

To "re-apply" the color the user want and override the underlying CSL